.github/workflows/workflow.yml

```diff
@@ -54,14 +55,15 @@ jobs:
       id: cache
       with:
         path: ~/.cache/pip
-        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/docs.txt', 'requirements/main.txt') }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/docs.txt', 'requirements/docs.txt') }}
         restore-keys: |
           ${{ runner.os }}-pip-
 
     - name: Install dependencies
       run: |
         pip install .
-        pip install -e .[docs]
+        pip install -r requirements/docs.txt
+        pip install -r requirements/tests.txt
```

Comment:
docs might be also building docs for the tests; if that's the case, could exclude the tests and also not install the deps here